elkscmd/basic/host-ibmpc.c

```diff
@@ -0,0 +1,359 @@
+/*
+ * Architecture Specific routines for CGA
+ * Sep 2024 Takahiro Yamada
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#if __ia16__
+#include <arch/io.h>
+#else
+#define inb(port)           0
+#define inw(port)           0
+#define outb(value,port)
+#define outw(value,port)
+#endif
+#include "host.h"
+#include "basic.h"
+
+#define VIDEO_05_G320x200   0x0005
+#define VIDEO_03_T80x25     0x0003
+
+extern FILE *outfile;
+
+static int gmode = 0;
+static int exit_on = 0;
+
+typedef struct {
+    int x;
+    int y;
+    int fgc;
+    int bgc;
+    int r;
+} xyc_t;
+
+static xyc_t gxyc = {0, 0, 7, 0, 1};
+
+#if __ia16__
+void int_10(unsigned int ax, unsigned int bx,
+            unsigned int cx, unsigned int dx)
+{
+    __asm__ volatile ("push %ds;"
+                      "push %es;"
+                      "push %bp;"
+                      "push %si;"
+                      "push %di;");
+    __asm__ volatile ("int $0x10;"
+                      :
+                      :"a" (ax), "b" (bx), "c" (cx), "d" (dx)
+                      :"memory", "cc");
+    __asm__ volatile ("pop %di;"
+                      "pop %si;"
+                      "pop %bp;"
+                      "pop %es;"
+                      "pop %ds;");
+}
+
+void memclrw(unsigned int offset, unsigned int seg, unsigned int count)
+{
+    __asm__ volatile ("push %ds;"
+                      "push %es;"
+                      "push %bp;"
+                      "push %si;"
+                      "push %di;");
+    __asm__ volatile ("mov %%ax,%%di;"
+                      "mov %%bx,%%es;"
+                      "xor %%ax,%%ax;"
+                      "cld;"
+                      "rep;"
+                      "stosw;"
+                      :
+                      :"a" (offset), "b" (seg), "c" (count)
+                      :"memory", "cc");
+    __asm__ volatile ("pop %di;"
+                      "pop %si;"
+                      "pop %bp;"
+                      "pop %es;"
+                      "pop %ds;");
+}
+#else
+void int_10(unsigned int ax, unsigned int bx,
+            unsigned int cx, unsigned int dx)
+{
+}
+
+void memclrw(unsigned int offset, seg_t seg, unsigned int count)
+{
+}
+#endif
+
+void host_digitalWrite(int pin,int state) {
+}
+
+int host_digitalRead(int pin) {
+    return 0;
+}
+
+int host_analogRead(int pin) {
+    return 0;
+}
+
+void host_pinMode(int pin,int mode) {
+}
+
+void host_mode(int mode) {
+    gmode = mode;
```

Comment:
I realize this file started by copying from host-stubs.c first, but I notice that we have two distinct C coding styles now going on in the same file with regards to the starting left brace; pick one and change the others, thank you.

Reply:
Sorry, I couldn't understand which one you are pointing.

Are you talking about
void int_10()
{
}
and others like
void host_mode(){
}

Reply:
> Sorry, I couldn't understand which one you are pointing.

Either one, I was just suggesting that all the function start braces should be the same way. My fault, as originally I quickly wrote host-stubs.c and used the first-brace-on-the-same-line style which makes for short code (since they were all empty) but is not very often used coding style.